plan.go

```diff
@@ -1,28 +1,35 @@
-package main
+package plan
 
 import (
-	"fmt"
 	"github.com/hashicorp/terraform/terraform"
-	"os"
+	"strings"
 )
 
-func main() {
-	file, err := os.Open("enablekms.plan")
-	if err != nil {
-		panic(err)
-	}
-
-	plan, err := terraform.ReadPlan(file)
-	if err != nil {
-		panic(err)
-	}
+const (
+	DiffInvalid = iota
+	DiffNone
+	DiffCreate
+	DiffUpdate
+	DiffDestroy
+	DiffDestroyCreate
+)
 
-	//fmt.Printf("plan: %s", plan.String())
+func IsDestroyingEBSVolume(pl *terraform.Plan) (bool, []string) {
+	var resourceNames []string
+	isDestroyed := false
+	//fmt.Printf("%s: %+v %+v\n", key, resource.ChangeType(), resource.Attributes)
 
-	for _, module := range plan.Diff.Modules {
+	for _, module := range pl.Diff.Modules {
 		for key, resource := range module.Resources {
-			fmt.Printf("%s: %+v %+v\n", key, resource.ChangeType(), resource.Attributes)
+			//	fmt.Printf("%s: %+v %+v\n", key, resource.ChangeType(), resource.Attributes)
+			switch resource.ChangeType() {
+			case DiffDestroy, DiffDestroyCreate:
+				if strings.Contains(strings.SplitAfter(key, ":")[0], "aws_ebs") {
```

Comment:
What is aws_ebs here?

The resource is called [aws_ebs_volume](https://www.terraform.io/docs/providers/aws/d/ebs_volume.html)

Reply:
There is also a chance that `split.SplitAfter` will have an empty result. Which then will crash at the `[0]`

Reply:
Looks like i added the aws_ebs_volume bit after i had already commited and pushed. 

If it was empty wouldn't that just mean that the resource isn't an ebs volume. Assuming that name of the resource is always the part of the key?.

Reply:
To be honest AFAIK a key looks like that:
```
key := "aws_ebs_volume.volume.0"
```
So I don't really understand why we splitting using `:`. The contains is really dirty as well. As it will match to widely.

We can also have a call and discuss that..


```